From 5278a77eacb35559b65aca656c060b42c1fe5926 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Fri, 28 Sep 2018 18:18:15 +0200
Subject: [PATCH] glslang: update for new glslang version

This adds some new members to this struct. I still have zero clue why
they don't provide this structure for us, so we don't have to copy/paste
it from example code..
---
 src/glsl/glslang.cc | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/glsl/glslang.cc b/src/glsl/glslang.cc
index d0fd081..8b98bf6 100644
--- a/src/glsl/glslang.cc
+++ b/src/glsl/glslang.cc
@@ -192,6 +192,18 @@ const TBuiltInResource DefaultTBuiltInResource = {
     /* .MaxCullDistances = */ 8,
     /* .MaxCombinedClipAndCullDistances = */ 8,
     /* .MaxSamples = */ 4,
+#if GLSLANG_PATCH_LEVEL >= 2892
+    /* .maxMeshOutputVerticesNV = */ 256,
+    /* .maxMeshOutputPrimitivesNV = */ 512,
+    /* .maxMeshWorkGroupSizeX_NV = */ 32,
+    /* .maxMeshWorkGroupSizeY_NV = */ 1,
+    /* .maxMeshWorkGroupSizeZ_NV = */ 1,
+    /* .maxTaskWorkGroupSizeX_NV = */ 32,
+    /* .maxTaskWorkGroupSizeY_NV = */ 1,
+    /* .maxTaskWorkGroupSizeZ_NV = */ 1,
+    /* .maxMeshViewCountNV = */ 4,
+#endif
+
     /* .limits = */ {
         /* .nonInductiveForLoops = */ 1,
         /* .whileLoops = */ 1,
